copy/copy.go

```diff
@@ -459,7 +478,9 @@ func (ic *imageCopier) copyLayer(srcInfo types.BlobInfo) (types.BlobInfo, digest
 			return types.BlobInfo{}, "", errors.Wrap(diffIDResult.err, "Error computing layer DiffID")
 		}
 		logrus.Debugf("Computed DiffID %s for layer %s", diffIDResult.digest, srcInfo.Digest)
+		ic.mu.Lock()
 		ic.cachedDiffIDs[srcInfo.Digest] = diffIDResult.digest
+		ic.mu.Unlock()
```

Comment:
Below this:
- The progress bar needs handling somehow.
- `ShouldCompressLayers()` needs auditing, or I guess preferably calling only once outside of the parallel code.
- `PutBlob` needs auditing.